### PR TITLE
Add MCV.js

### PR DESCRIPTION
**Pull request by owcz**
Translator for https://www.mcvuk.com

### CODE REVIEW

File: MCV.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+
+// attr()/text() v2 per https://github.com/zotero/translators/issues/1277
+function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
+
+
+function detectWeb(doc, url) {
+	if (/\/(business|development|esports|influence)\//.test(url)) {
+		return "blogPost";
+	} else if (url.indexOf("/search?query=") != -1) {
```
Comment (adam3smith) [inline MCV.js]: use `string.includes()` for string testing & take advantage of `getSearchResults`

File: MCV.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+		return "multiple";
+	} else return null;
```
Comment (adam3smith) [inline MCV.js]: no need for the `else`; that's implicit.

File: MCV.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+	var translator = Zotero.loadTranslator('web');
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48'); // embedded metadata
+	translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		item.itemType = "blogPost";
+		item.publicationTitle = "MCV";
+		item.date = attr(doc,'meta[name="published"]','content');
+		item.title = item.title.replace(/- MCV$/,'');
+		if (item.creators[0].lastName == "Editors") {
```
Comment (adam3smith) [inline MCV.js]: just to make sure -- this would always be the first author? No need to cycle through creatores?

Comment (owcz) [reply MCV.js]: yep this is just so the generic "MCV Editors" byline doesn't get digested into "Editors, MCV"

don't believe I've seen two+ authors on this site, but the generic byline wouldn't be used with a second author, fwiw

**Comment by owcz** [comment]
good point re: type
I didn't know if all or just some of their content was also in their magazine
but changed the type anyway

File: MCV.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+*/
+
+
+// attr()/text() v2 per https://github.com/zotero/translators/issues/1277
+function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
+
+
+function detectWeb(doc, url) {
+	if (/\/(business|development|esports|influence)\//.test(url)) {
+		return "blogPost";
```
Comment (adam3smith) [inline MCV.js]: update the item type here and in `doWeb` and update tests

File: MCV.js [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48'); // embedded metadata
+	translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		item.itemType = "magazineArticle";
+		item.publicationTitle = "MCV";
+		item.ISSN = "1469-4832";
+		item.date = attr(doc,'meta[name="published"]','content');
+		item.title = item.title.replace(/- MCV$/,'');
+		item.language = item.language.replace('us','US');
```
Comment (adam3smith) [inline MCV.js]: just in case let's wrap this into an `if (item.language)` so it doesn't break things if they remove language or it's missing.

**Pull request merged**